auth/auth.go

```diff
@@ -0,0 +1,27 @@
+package auth
+
+type key string
```

Comment:
actually had to create own type because of this static check error:
> should not use built-in type string as key for value; define your own type to avoid collisions (SA1029) go-staticcheck